status/condition_set_test.go

```diff
@@ -105,4 +107,63 @@ var _ = Describe("Conditions", func() {
 		Expect(testObject.StatusConditions().IsTrue(ConditionTypeFoo, ConditionTypeBaz)).To(BeTrue())
 		Expect(testObject.StatusConditions().IsTrue(ConditionTypeFoo, ConditionTypeBar, ConditionTypeBaz)).To(BeTrue())
 	})
+	It("should return nil when status conditions are not found on the expected path", func() {
+		testObject := &unstructured.Unstructured{Object: map[string]interface{}{
+			"invalid": map[string]interface{}{
+				"invalid": "invalid",
+			},
+		}}
+		obj, err := status.FromUnstructured(testObject)
+		Expect(obj).To(BeNil())
+		Expect(err).To(HaveOccurred())
+	})
+	It("should validate status condition on unstructured object status is false", func() {
+		unstructuredAdapter, err := status.FromUnstructured(createUnstructuredStatusConditions("False"))
+		Expect(err).To(BeNil())
+		Expect(unstructuredAdapter).To(Not(BeNil()))
+		Expect(unstructuredAdapter.StatusConditions().IsTrue(status.ConditionReady)).To(BeFalse())
+	})
+	It("should validate status condition on unstructured object status is true", func() {
+		unstructuredAdapter, err := status.FromUnstructured(createUnstructuredStatusConditions("True"))
+		Expect(err).To(BeNil())
+		Expect(unstructuredAdapter).To(Not(BeNil()))
+		Expect(unstructuredAdapter.StatusConditions().IsTrue(status.ConditionReady)).To(BeTrue())
+	})
+	It("should set condition on unstructured object", func() {
+		testObject := &unstructured.Unstructured{Object: map[string]interface{}{
+			"status": map[string]interface{}{
+				"conditions": []interface{}{},
+			},
+		}}
+		conditions := []status.Condition{
+			{
+				Type:    status.ConditionSucceeded,
+				Status:  metav1.ConditionFalse,
+				Reason:  "reason",
+				Message: "message",
+			},
+		}
+		unstructuredAdapter, err := status.FromUnstructured(testObject)
+		Expect(err).To(BeNil())
```

Comment:
```suggestion
		Expect(err).ToNot(HaveOccurred())
```

nit: You seem to use this style elsewhere, so consider keeping it for consistency